src/pages/login/login.test.tsx

```diff
@@ -0,0 +1,139 @@
+
+import { render, fireEvent, waitFor ,screen} from '@testing-library/react';
+import { Login } from './login';
+
+import { Provider } from 'react-redux';
+import { store } from '../../store/store';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+describe('Login component', () => {
+  it('renders login form initially', async () => {
+    const mockNavigate = jest.fn();
+    const { getByText, getByLabelText } = render(
+      <Provider store={store}>
+        <Router>
+          <Login navigate={mockNavigate} />
+        </Router>
+      </Provider>
+    );
+    
+    //expect(getByText('Login')).toBeInTheDocument();
+    expect(getByLabelText('Email:')).toBeInTheDocument();
+    expect(getByLabelText('Password:')).toBeInTheDocument();
+  });
+
+  it('validates email', async () => {
+    const mockNavigate = jest.fn();
+    const { getByText, getByLabelText, getByRole } = render(
+      <Provider store={store}>
+        <Router>
+          <Login navigate={mockNavigate} />
+        </Router>
+      </Provider>
+    );
+
+    const emailInput = getByLabelText('Email:');
+    const passwordInput = getByLabelText('Password:');
+   
+
+    fireEvent.change(emailInput, { target: { value: 'notanemail' } });
+    //fireEvent.change(passwordInput, { target: { value: 'short' } });
+
+    const loginButton =getByRole("button",{name: /Login/i})
+    await fireEvent.click(loginButton);
+
+    const emailErrorElement = await screen.findByText(
+        "Please enter valid email !"
+      );
+      expect(emailErrorElement).toBeInTheDocument();
+    
+  });
+
+  it('validates password', async () => {
```

Comment:
i dont think desc what it does, please look into naming your tests to what it does